posthog/management/commands/generate_source_configs.py

```diff
@@ -0,0 +1,393 @@
+import logging
+from structlog import get_logger
+from temporalio import workflow
+
+with workflow.unsafe.imports_passed_through():
+    from django.core.management.base import BaseCommand
+
+import os
+from typing import Any, Optional
+
+from posthog.warehouse.api.available_sources import AVAILABLE_SOURCES
+from posthog.warehouse.models import ExternalDataSource
+from posthog.schema import (
+    SourceConfig,
+    SourceFieldInputConfig,
+    SourceFieldSelectConfig,
+    SourceFieldSwitchGroupConfig,
+    SourceFieldOauthConfig,
+    SourceFieldFileUploadConfig,
+    SourceFieldSSHTunnelConfig,
+    Type4,
+)
+
+logger = get_logger(__name__)
+logger.setLevel(logging.INFO)
+
```

Comment:
I'd add a comment/blurb on this  or a readme somewhere that describes how the internals are linked. We shouldn't need to interact with this often but that means at some point if we do a reference would be nice